banking-app/Makefile

```diff
@@ -10,23 +10,28 @@ build: ## 🔨 Build the Banking Application
 	@echo -e "\e[34m$@\e[0m" || true
 	@npm run build
 
-test: ## 🧪 Test the Banking Application
+test: build ## 🧪 Test the Banking Application in the sandbox
 	@echo -e "\e[34m$@\e[0m" || true
-	@./test.sh
+	@. ./scripts/test_sandbox.sh --nodeAddress 127.0.0.1:8000 --certificate_dir ./workspace/sandbox_common
+
+test-docker-virtual: build-virtual ## 🧪 Test the Banking Application in a Docker sandbox
+	@echo -e "\e[34m$@\e[0m" || true
+	@. ./scripts/test_docker.sh --virtual --serverIP 172.17.0.3 --port 8080
+
+test-docker-enclave: build-enclave ## 🧪 Test the Banking Application in a Docker enclave
+	@echo -e "\e[34m$@\e[0m" || true
+	@. ./scripts/test_docker.sh --enclave --serverIP 172.17.0.4 --port 8080
 
-# Hack to use default resolve.js
-# It should be improved after '--no-default-constitution' is added to sandbox.sh.
-# NOTE: This hack replaces the content of /opt/ccf/bin/resolve.js
 # Run sandbox. Consider 3 members as 3 banks.
-start-host: build ## 🏃 Start the CCF Sandbox
+# This is used in the demo scripts
+start-host: build ## 🏁 Start the CCF Sandbox for the demo
 	@echo -e "\e[34m$@\e[0m" || true
-	@cp constitutions/resolve.js /opt/ccf/bin/resolve.js
-	@/opt/ccf/bin/sandbox.sh --js-app-bundle ./dist/ --initial-member-count 3 --initial-user-count 0
+	@/opt/ccf/bin/sandbox.sh --js-app-bundle ./dist/ --initial-member-count 3 --initial-user-count 0 --constitution-dir ./constitution
 
 demo: ## 🎬 Demo the Banking Application
```

Comment:
`make demo` throw exception "Failed to connect to 127.0.0.1 port 8000: Connection refused"

Reply:
This is because when you "demo" you need to run `make start-host` in another terminal. I have added a check to see if the sandbox is running and tell the user.

make test - creates and tears down each time
make demo - is an interactive session with a network.